apps/client/src/routes/courses/[slug]/+layout.server.ts

```diff
@@ -0,0 +1,36 @@
+import type { LayoutServerLoad } from "./$types";
+import {
+  FETCH_COURSE_BY_SLUG,
+  FETCH_USER_COURSE_PROGRESS,
+} from "$lib/graphql/queries/courses";
+import apolloClient from "$lib/graphql/apolloClient";
+
+export const load: LayoutServerLoad = async ({ params, parent }) => {
+  const { session } = await parent();
+
+  const { slug } = params;
+
+  const getCourseData = async () => {
```

Comment:
Could you define these functions outside of the body of this function? We intend to export these functions for unit tests.

Reply:
Will do